flashloaders/Makefile

```diff
@@ -4,35 +4,60 @@
 # This makefile will save your time from dealing with compile errors
 # Adjust CC if needed
 
-CC = /opt/local/gcc-arm-none-eabi-8-2018-q4-major/bin/arm-none-eabi-gcc
-
-CFLAGS_thumb1 = -mcpu=Cortex-M0  -Tlinker.ld -ffreestanding -nostdlib
-CFLAGS_thumb2 = -mcpu=Cortex-M3  -Tlinker.ld -ffreestanding -nostdlib
-
-all: stm32vl.o stm32f0.o stm32l.o stm32f4.o stm32f4_lv.o stm32l4.o stm32f7.o stm32f7_lv.o
-
-stm32vl.o: stm32f0.s
-	$(CC) stm32f0.s $(CFLAGS_thumb2)  -o stm32vl.o
-stm32f0.o: stm32f0.s
-	$(CC) stm32f0.s $(CFLAGS_thumb1)  -o stm32f0.o
-stm32l.o: stm32lx.s
-	$(CC) stm32lx.s $(CFLAGS_thumb2)  -o stm32l.o
-stm32f4.o: stm32f4.s
-	$(CC) stm32f4.s $(CFLAGS_thumb2)  -o stm32f4.o
-stm32f4_lv.o: stm32f4lv.s
-	$(CC) stm32f4lv.s $(CFLAGS_thumb2)  -o stm32f4_lv.o
-stm32l4.o: stm32l4.s
-	$(CC) stm32l4.s $(CFLAGS_thumb2)  -o stm32l4.o
-stm32f7.o: stm32f7.s
-	$(CC) stm32f7.s $(CFLAGS_thumb2)  -o stm32f7.o
-stm32f7_lv.o: stm32f7lv.s
-	$(CC) stm32f7lv.s $(CFLAGS_thumb2)  -o stm32f7_lv.o
+CROSS_COMPILE ?= /opt/gcc-arm-none-eabi-6-2017-q2-update/bin/arm-none-eabi-
 
-clean:
-	rm *.o
+CC = $(CROSS_COMPILE)gcc
+OBJCOPY = $(CROSS_COMPILE)objcopy
+
+XXD = xxd
+XXDFLAGS = -i -c 4 
+
+CFLAGS_THUMB1 = -mcpu=Cortex-M0 -Tlinker.ld -ffreestanding -nostdlib
```

Comment:
IMO ARM{v6M/v7M}  [ _BASELINE ]  is more correct than THUMB1/2

Reply:
Yes, it will be more informative